### PR TITLE
Add additional logging in case of Pension errors

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Summary

In order to more effectively catch and debug errors, adds additional logs in rescue blocks, and add additional StatsD metrics.

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/77412

## Testing done

- [x] New code is covered by unit tests
- [x] Raised errors in newly logged methods and tested that when running the app the errors were logged

## Screenshots

Screenshot 2024-03-25 at 11 12 47 AM
Screenshot 2024-03-25 at 11 30 03 AM
Screenshot 2024-03-25 at 11 37 03 AM

## What areas of the site does it impact?
Pension Benefits

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/controllers/v0/pension_claims_controller.rb [hunk 1]
```diff
@@ -12,40 +12,31 @@ def claim_class
...
-              attempt_id: submission_attempt.id,
-              aasm_state: submission_attempt.aasm_state
-            }
-          }
-        }
+        response = format_show_response(claim, state, form_submission, submission_attempt)
       end
-
       render(json: response)
+    rescue ActiveRecord::RecordNotFound => e
```
Comment (mjknight50) [inline app/controllers/v0/pension_claims_controller.rb:25]: Does Ruby does require some form of “begin” to appear before the rescue?

**Review by mjknight50 — CHANGES_REQUESTED** [review]

File: app/controllers/v0/pension_claims_controller.rb (hunk 1 above)
Comment (TaiWilkin) [reply app/controllers/v0/pension_claims_controller.rb:25]: A method 'def' can serve as a begin - so in this case, the rescue looks for any non-caught errors inside the method

**Review by mjknight50 — DISMISSED** [review]
Looks good!

File: app/controllers/v0/pension_claims_controller.rb (hunk 1 above)
Comment (mjknight50) [reply app/controllers/v0/pension_claims_controller.rb:25]: Thank you for explaining it.

**Comment by mjknight50** [comment]
One comment outside of the approval is that we should fix the .codeowners and remove bananapeels

File: app/controllers/v0/pension_claims_controller.rb [hunk 2]
```diff
@@ -81,5 +71,37 @@ def log_validation_error_to_metadata(in_progress_form, claim)
       metadata['submission']['error_message'] = claim&.errors&.errors&.to_s
       in_progress_form.update(metadata:)
     end
+
+    def format_show_response(claim, state, form_submission, submission_attempt)
```
Comment (LindseySaari) [inline app/controllers/v0/pension_claims_controller.rb:75]: These methods don't seem like they should belong in the control. To keep things clean, could you move them to a helper/module?

**Review by wayne-weibel — APPROVED** [review]

**Review by mathisto — APPROVED** [review]
happy-college-basketball-gif-by-wisconsin-badgers

**Review by mjknight50 — APPROVED** [review]
LGTM

**Review by mathisto — DISMISSED** [review]

File: app/controllers/v0/pension_claims_controller.rb [hunk 3]
```diff
@@ -1,11 +1,14 @@
 # frozen_string_literal: true
 
 require 'pension_21p527ez/tag_sentry'
+require 'pension_21p527ez/monitor'
 
 module V0
   class PensionClaimsController < ClaimsBaseController
     service_tag 'pension-application'
 
+    MONITOR = ::Pension21p527ez::Monitor.new
```
Comment (stevenjcumming) [inline app/controllers/v0/pension_claims_controller.rb:10]: The scope resolution operator `::` isn't necessary. This should also be moved to a private method such as
```
def pension_monitor
  Pension21p527ez::Monitor.new
 end 
```

File: app/controllers/v0/pension_claims_controller.rb [hunk 4]
```diff
@@ -14,64 +17,50 @@ def claim_class
       SavedClaim::Pension
     end
 
-    # rubocop:disable Metrics/MethodLength
     def show
       claim = claim_class.find_by!({ guid: params[:id] }) # will raise ActiveRecord::NotFound
```
Comment (stevenjcumming) [inline app/controllers/v0/pension_claims_controller.rb:21]: Consider updating `find_by!({ guid: params[:id] })` to `find_by!(guid: params[:id])`

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**